src/hotspot/share/gc/shared/cardTableBarrierSet.hpp

```diff
@@ -60,10 +60,11 @@ class CardTableBarrierSet: public ModRefBarrierSet {
                       CardTable* card_table,
                       const BarrierSet::FakeRtti& fake_rtti);
 
- public:
-  CardTableBarrierSet(CardTable* card_table);
   ~CardTableBarrierSet();
 
+public:
```

Comment:
The destructor should remain public - this class is directly instantiated.
It should also be virtual, since this class is also used as a base class.  This is a small but typical example
of why classes shouldn't do such double duty.